scripts/generateMetadata.js

```diff
@@ -46,32 +53,55 @@ async function docgen(filePath) {
   }
 }
 
-const getComponentNameFromFile = (file) => file.replace(/^.*[\\/]/, '').replace(/\.js/, '');
+const getComponentNameFromFile = (file) => path.basename(file).replace(/\.js/, '');
 
-// Get files recursively as an array
-const getFilesFromDirectory = (directoryPath) => {
-  const filesInDirectory = fs.readdirSync(path.join(root, directoryPath));
-  const files = filesInDirectory.map((file) => {
-    const filePath = path.join(directoryPath, file);
-    const stats = fs.statSync(path.join(root, filePath));
+/** Analyzes subcomponents of a component in given path and extracts paths they're imported from. */
+function getSubcomponentPaths(componentPath) {
+  const componentName = getComponentNameFromFile(componentPath);
 
-    if (stats.isDirectory()) {
-      return getFilesFromDirectory(filePath);
-    }
+  /** Matches lines that are like `Component.Subcomponent = Subcomponent` */
+  const subcomponentRegExp = new RegExp(
+    `(${componentName}\\.[A-Z]\\w+) = (?<subcomponentName>\\w+)`,
+    'g',
+  );
 
-    return filePath;
-  });
+  const fileContent = fs.readFileSync(path.join(root, componentPath), 'utf-8');
+  const subcomponentNameMatches = fileContent.matchAll(subcomponentRegExp);
+  const subcomponentNames = [...subcomponentNameMatches].map((a) => a.groups.subcomponentName);
+
+  if (!subcomponentNames.length) return [];
+
+  /** Matches lines that are like `Subcomponent from './path/to/subcomponent';` */
+  const subcomponentPathRegExp = new RegExp(
+    `(${subcomponentNames.join('|')}) from '(?<path>[.\\w\\/-]+)';`,
+    'g',
+  );
+
+  const subcomponentPathMatches = fileContent.matchAll(subcomponentPathRegExp);
+  const subcomponentPaths = [...subcomponentPathMatches].map((match) =>
+    path.join(componentPath, '../', match.groups.path),
+  );
 
-  return files.filter((file) => file.length).flat();
-};
+  return subcomponentPaths;
+}
+
+/**
+ * Gets components that are imported in `index.js` of given directory (assuming they're all exported).
```

Comment:
One caveat is that we cannot have new component pages that aren't listed in the index. Now for development we must add them to the index file. 

I don't think it's a big issues seeing how we work, but good having that in mind when setting up new components. Maybe add a comment to `scripts/templates/README.md`

Reply:
@AlbertCarreras I don't think it is a caveat. `generateComponent` script automatically adds new components to `index.js`. We import components from `index.js` to the docs site, we never import from the component file directly.

Though, I agree adding a comment to README